datastore/google/cloud/datastore/query.py

```diff
@@ -441,43 +441,33 @@ def _build_protobuf(self):
 
         return pb
 
-    def _process_query_results(self, entity_pbs, cursor_as_bytes,
-                               more_results_enum, skipped_results):
+    def _process_query_results(self, response_pb):
         """Process the response from a datastore query.
 
-        :type entity_pbs: iterable
-        :param entity_pbs: The entities returned in the current page.
-
-        :type cursor_as_bytes: bytes
-        :param cursor_as_bytes: The end cursor of the query.
-
-        :type more_results_enum:
-            :class:`.query_pb2.QueryResultBatch.MoreResultsType`
-        :param more_results_enum: Enum indicating if there are more results.
-
-        :type skipped_results: int
-        :param skipped_results: The number of skipped results.
+        :type response_pb: :class:`.datastore_pb2.RunQueryResponse`
+        :param response_pb: The protobuf response from a ``runQuery`` request.
 
         :rtype: iterable
         :returns: The next page of entity results.
         :raises ValueError: If ``more_results`` is an unexpected value.
         """
-        self._skipped_results = skipped_results
+        self._skipped_results = response_pb.batch.skipped_results
 
-        if cursor_as_bytes == b'':  # Empty-value for bytes.
+        if response_pb.batch.end_cursor == b'':  # Empty-value for bytes.
```

Comment:
@daspecster 

1. The comment is intended to explain why the check equal to `b''` rather than using `HasField`. This is because for "scalar types", `HasField` doesn't work in pb objects (it throws an exception). Instead we equate being set to the "empty-value" to be unset.
1. The comment was pre-existing